astropy/coordinates/builtin_frames/altaz.py

```diff
@@ -77,13 +91,35 @@ class AltAz(BaseCoordinateFrame):
     """
 
     frame_specific_representation_info = {
-        'spherical': [RepresentationMapping('lon', 'az'),
-                      RepresentationMapping('lat', 'alt')],
+        r.SphericalRepresentation: [
+            RepresentationMapping('lon', 'az'),
+            RepresentationMapping('lat', 'alt')
+        ],
+        r.SphericalCosLatDifferential: [
+            RepresentationMapping('d_lon_coslat', 'pm_az', u.mas/u.yr),
```

Comment:
The test case I meant was iniatializing - presumably, as it was, it would fail with `AltAz(..., pm_az_cosalt=..., ...)`.